src/memoryModelsRepo/overwritePersonaAttribute.ts

```diff
@@ -0,0 +1,59 @@
+import { assign, findIndex } from 'lodash';
+import { v4 as uuid } from 'uuid';
+import Attribute from '../models/Attribute';
+import OverwritePersonaAttributeOptions from // tslint:disable-line:import-spacing
+  '../repoFactory/options/OverwritePersonaAttributeOptions';
+import OverwritePersonaAttributeResult from // tslint:disable-line:import-spacing
+  '../repoFactory/results/OverwritePersonaAttributeResult';
+// tslint:disable-next-line:no-unused
+import _OverwritePersonaAttributeOptions from // tslint:disable-line:import-spacing
+  '../serviceFactory/options/OverwritePersonaAttributeOptions';
+// tslint:disable-next-line:no-unused
+import _OverwritePersonaAttributeResult from // tslint:disable-line:import-spacing
+  '../serviceFactory/results/OverwritePersonaAttributeResult';
+import Config from './Config';
+
+export default (config: Config) => {
+  return async ({
+    key,
+    organisation,
+    personaId,
+    value,
+  }: OverwritePersonaAttributeOptions): Promise<OverwritePersonaAttributeResult> => {
+      const personaAttributeIndex = findIndex(config.state.personaAttributes, {
```

Comment:
In general the behaviour of this function is a little unclear. Let's create a boolean variable here called `isAlreadyStored`, then split this function into two branches of an if statement. In the first branch create a new Attribute and append. In the second branch update the existing Attribute.